### ISSUE TITLE
doesn't run

### ISSUE THREAD

**Issue by damianatorrpm** [flathub/ca.littlesvr.asunder#19]
(asunder:2): Gtk-WARNING **: 06:00:23.294: cannot open display:

**Comment by Eonfge** [comment]
Can you provide me some more info? Run the following commands and tell me what they return:

```
inxi -Fx
```

```
flatpak info ca.littlesvr.asunder
```

**Comment by damianatorrpm** [comment]
Note that this is on wayland.

```
$ inxi -Fx
System:    Host: fedora Kernel: 5.9.13-200.fc33.x86_64 x86_64 bits: 64 compiler: gcc v: 2.35-15.fc33) Desktop: GNOME 3.38.2 
           Distro: Fedora release 33 
Machine:   Type: Desktop System: Gigabyte product: N/A v: N/A serial: <superuser/root required> 
           Mobo: Gigabyte model: 990XA-UD3 serial: <superuser/root required> UEFI: American Megatrends v: FEh date: 10/29/2014 
CPU:       Info: 8-Core model: AMD FX-8320 bits: 64 type: MCP arch: Bulldozer L2 cache: 2048 KiB 
           flags: avx lm nx pae sse sse2 sse3 sse4_1 sse4_2 sse4a ssse3 svm bogomips: 56256 
           Speed: 1405 MHz min/max: 1400/3500 MHz boost: enabled Core speeds (MHz): 1: 1405 2: 1391 3: 1402 4: 1402 5: 3103 
           6: 3693 7: 1403 8: 1401 
Graphics:  Device-1: Advanced Micro Devices [AMD/ATI] Ellesmere [Radeon RX 470/480/570/570X/580/580X/590] 
           vendor: Sapphire Limited driver: amdgpu v: kernel bus ID: 01:00.0 
           Display: wayland server: Fedora Project X.org 1.20.10 compositor: gnome-shell driver: amdgpu 
           note: display driver n/a - try sudo/root resolution: 1: 1920x1080~60Hz 2: 1920x1080~60Hz 3: 1920x1080~60Hz 
           OpenGL: renderer: Radeon RX 570 Series (POLARIS10 DRM 3.39.0 5.9.13-200.fc33.x86_64 LLVM 11.0.0) v: 4.6 Mesa 20.2.4 
           direct render: Yes 
Audio:     Device-1: Advanced Micro Devices [AMD/ATI] SBx00 Azalia vendor: Gigabyte driver: snd_hda_intel v: kernel 
           bus ID: 00:14.2 
           Device-2: AMD Ellesmere HDMI Audio [Radeon RX 470/480 / 570/580/590] vendor: Sapphire Limited driver: snd_hda_intel 
           v: kernel bus ID: 01:00.1 
           Sound Server: ALSA v: k5.9.13-200.fc33.x86_64 
Network:   Device-1: Realtek RTL8111/8168/8411 PCI Express Gigabit Ethernet vendor: Gigabyte driver: r8169 v: kernel 
           port: d000 bus ID: 03:00.0 
           IF: enp3s0 state: up speed: 100 Mbps duplex: full mac: 94:de:80:b7:a4:a7 
Drives:    Local Storage: total: 1.27 TiB used: 948.22 GiB (72.7%) 
           ID-1: /dev/sda vendor: Samsung model: SSD 840 EVO 120GB size: 111.79 GiB temp: 25 C 
           ID-2: /dev/sdb vendor: TeamGroup model: T253X2001T size: 953.87 GiB 
           ID-3: /dev/sdc vendor: Samsung model: SSD 850 PRO 256GB size: 238.47 GiB temp: 31 C 
Partition: ID-1: /boot size: 975.9 MiB used: 90.4 MiB (9.3%) fs: ext4 dev: /dev/sdb2 
           ID-2: /usr size: 952.28 GiB used: 363.82 GiB (38.2%) fs: overlay source: ERR-102 
Swap:      Alert: No Swap data was found. 
Sensors:   System Temperatures: cpu: 35.6 C mobo: N/A gpu: amdgpu temp: 35.0 C 
           Fan Speeds (RPM): N/A gpu: amdgpu fan: 978 
sh: line 1: 159818 Segmentation fault      (core dumped) /bin/flatpak list 2> /dev/null
Info:      Processes: 411 Uptime: 9h 56m Memory: 15.58 GiB used: 5.19 GiB (33.3%) Init: systemd runlevel: 5 Compilers: 
           gcc: 10.2.1 clang: 11.0.0 Packages: 1900 Shell: Bash v: 5.0.17 inxi: 3.1.08 
```

```
$ flatpak info ca.littlesvr.asunder

Asunder CD Ripper - Asunder is a graphical Audio CD ripper and encoder for
Linux. You can use it to save tracks from an Audio CD as any of WAV, MP3, OGG,
FLAC, Opus, WavPack, Musepack, AAC, and Monkey's Audio files.

          ID: ca.littlesvr.asunder
         Ref: app/ca.littlesvr.asunder/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: 2.9.6
     License: GPL-2.0-only
      Origin: flathub
  Collection: org.flathub.Stable
Installation: user
   Installed: 10.3 MB
     Runtime: org.freedesktop.Platform/x86_64/20.08
         Sdk: org.freedesktop.Sdk/x86_64/20.08

      Commit: 675148df469fa6e7966aefcb61fe85d5061e2a11be778331e82e2594be684bce
      Parent: 71fdde1e112011ca255af68755c02b2f43f082d7d3dde2e29e3b20080f32cae5
     Subject: Update shared moduels for MAC Dependency (a5b3c92d)
        Date: 2020-12-16 21:05:42 +0000
```

```
$ flatpak run ca.littlesvr.asunder
(asunder:2): Gtk-WARNING **: 18:28:58.841: cannot open display: 
```

**Comment by damianatorrpm** [comment]
Aso note that flatpak is git latest.

**Comment by Eonfge** [comment]
I have been able to reproduce this in a Boxes VM. Guess it's something related to GTK 2.

Some further questions if you don't mind:
- Has the application ever worked?
- So yes, what was the last time you used it?
- Have you tried it on X11?

**Comment by damianatorrpm** [comment]
>Has the application ever worked?
No.

>So yes, what was the last time you used it?
No.

>Have you tried it on X11?
No.

I have been testing applications from flathub on wayfire wayland compositor.
(and asunder ist the only application that fails - except from ones requiring gnome shell dbus - Almond and Hack)
I reported this here since it fails on gnome as well.

**Comment by Eonfge** [comment]
Resolved with #20
